README.md

```diff
@@ -173,11 +173,57 @@ Using the `Parameters` classes is not required, but they can help in a number of
 - Provides a more natural way to construct parameters
 - Facilitates ASP.NET Core model binding (bind an HTML form to a `Parameters` instance)
 
+### HTTP Hooks
+
+Users can audit the HTTP requests and responses being made by the library by setting the `Hooks` property of a `ClientConfiguration` with a set of event handlers. Available handlers include:
+
+- `OnRequestExecuting` - Called before an HTTP request is made. An `OnRequestExecutingEventArgs` object is passed to the
+  handler, which contains details about the `HttpRequestMessage` that will be sent to the server.
+  - The `HttpRequestMessage` at this point is configured with all expected data (headers, body, etc.). Modifying
+    any data in the callback will NOT affect the actual request that is sent to the server.
+- `OnRequestResponseReceived` - Called after an HTTP request is made. An `RequestResponseReceivedEventArgs` object is
+  passed to the handler, which contains details about the `HttpResponseMessage` that was received from the server.
+
+Users can interact with these details in their callbacks as they see fit (e.g. logging).
+
+```csharp
+void OnRequestExecutingHandler(object? sender, OnRequestExecutingEventArgs args) {
+    // Interact with the HttpRequestMessage here via args
```

Comment:
I would modify this comment a little bit since the args have more than the `HttpRequestMessage` attribute. Same thing down below